File: Difficulty: Easy/Nth number made of prime digits/nth-number-made-of-prime-digits.java
//{ Driver Code Starts
// Initial Template for Java

/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        while (t-- > 0) {
            int n = sc.nextInt();
            Solution ob = new Solution();
            System.out.println(ob.primeDigits(n));

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    // Function to find nth number made of only prime digits.
    public static int primeDigits(int n) {
        int count = 0, number = 1;
        
        while (count < n) {
            if (hasOnlyPrimeDigits(number)) {
                count++;
            }
            number++;
        }
        return number - 1;
    }
    
    private static boolean hasOnlyPrimeDigits(int num) {
        while (num != 0) {
            int digit = num % 10;
            if (digit != 2 && digit != 3 && digit != 5 && digit != 7) {
                return false;
            }
            num /= 10;
        }
        return true;
    }
}

File: Difficulty: Medium/Nth number made of prime digits/nth-number-made-of-prime-digits.py
from collections import deque

class Solution:
    def primeDigits(self, n):
        q = deque()
        q.append(2)
        q.append(3)
        q.append(5)
        q.append(7)
        
        count, result = 0, 0
        
        while count < n:
            result = q.popleft()
            count += 1
            
            if count == n:
                return result
            
            q.append(result * 10 + 2)
            q.append(result * 10 + 3)
            q.append(result * 10 + 5)
            q.append(result * 10 + 7)
        
        return result